device.c

```diff
@@ -62,25 +63,49 @@ static void device_lock(struct device *device)
 	if (fd >= 0)
 		close(fd);
 
-	fd = open(lock, O_RDONLY | O_CLOEXEC);
+	fd = open(lock, O_RDWR | O_CLOEXEC);
 	if (fd < 0)
 		err(1, "failed to open lockfile %s", lock);
 
+	/* Read current user out of the lockfile if there is one */
+	n = read(fd, user, sizeof(user)-1);
+	if (n < 0)
+		err(1, "failed to read lockfile %s", lock);
+	/* Strip newline */
+	if (n)
+		user[n-1] = '\0';
+
 	while (1) {
 		char c;
 
 		n = flock(fd, LOCK_EX | LOCK_NB);
 		if (!n)
-			return;
+			break;
 
-		warnx("board is in use, waiting...");
+		warnx("board is in use by %s, waiting...", user);
 
 		sleep(3);
 
 		/* check that connection isn't gone */
 		if (read(STDIN_FILENO, &c, 1) == 0)
 			errx(1, "connection is gone");
 	}
+
+	/* Write our username to the lockfile */
+	n = snprintf(user, sizeof(user), "%s\n", cdba_user);
```

Comment:
I think skipping `\n` might be a better choice, it saves the code on both read and write sides.